SimpleRabbit.NetCore/Service/BasicRabbitService.cs

```diff
@@ -29,18 +27,18 @@ protected ConnectionFactory Factory
                     return _factory;
                 }
 
-                var config = _config.Get(ConfigurationName);
+                var config = _config;
 
-                if (config?.Uri == null)
+                if (config?.Username == null || config.Password == null)
                 {
-                    throw new ArgumentNullException(nameof(config.Uri), "Rabbit configuration possible not set correctly.");
+                    throw new ArgumentNullException(nameof(_config), "Rabbit configuration possible not set correctly.");

```

Comment:
the second conjuctive should also be coalescing, config?.Password

Reply:
No need, as it will fail the first check if config is null already (and the second check will not be carried out).